build.rs

```diff
@@ -80,11 +84,194 @@ fn main() -> Result<()> {
     write_known_targets(out_dir)?;
     compile_syntax(out_dir).context("could not compile syntax files")?;
 
+    println!("cargo::rustc-check-cfg=cfg(icons_out_dir)");
+    println!("cargo:rustc-cfg=icons_out_dir");
+
+    let out_css = "static/icons.css";
+    let out_css = if std::env::var("RUN_IN_DOCKER").is_ok() {
+        let _ = std::fs::create_dir_all("/srv/docsrs/static");
+        Path::new("/srv/docsrs/").join(out_css)
+    } else {
+        let package_dir = env::var("CARGO_MANIFEST_DIR").context("missing CARGO_MANIFEST_DIR")?;
+        Path::new(&package_dir).join(out_css)
+    };
+    generate_css_icons(out_css, out_dir)?;
+
     // trigger recompilation when a new migration is added
     println!("cargo:rerun-if-changed=migrations");
     Ok(())
 }
 
+fn capitalize(s: &str) -> String {
+    let mut c = s.chars();
+    match c.next() {
+        None => String::new(),
+        Some(f) => f.to_uppercase().chain(c).collect(),
+    }
+}
+
+fn render_icon(
+    icon_name: &str,
+    icon_str: &str,
+    type_name: String,
+    code_output: &mut String,
+    css_output: &mut String,
+    icon_kind: &str,
+) {
+    let css_class = format!("f-a_{icon_name}_{icon_kind}");
+    css_output.push_str(&format!(
+        "\
+.{css_class} {{
```

Comment:
what I wonder: 

isn't a CSS distribution with the icons something that font-awesome already provides? That we can just use?